File: src/main/java/no/nav/familie/integrasjoner/journalpost/JournalpostRestClientException.java
package no.nav.familie.integrasjoner.journalpost;


public class JournalpostRestClientException extends RuntimeException {

    private String journalpostId;


    public JournalpostRestClientException(String message, Throwable cause, String journalpostId) {
        super(message, cause);
        this.journalpostId = journalpostId;
    }

    public String getJournalpostId() {
        return journalpostId;
    }
}

File: src/main/java/no/nav/familie/integrasjoner/journalpost/JournalpostRestClientException.kt
package no.nav.familie.integrasjoner.journalpost

import no.nav.familie.integrasjoner.journalpost.internal.SafJournalpostRequest

class JournalpostRestClientException(
    message: String?,
    cause: Throwable?,
    val journalpostId: String,
) : RuntimeException(message, cause)

class JournalpostRequestException(
    message: String?,
    cause: Throwable?,
    val safJournalpostRequest: SafJournalpostRequest,
) : RuntimeException(message, cause)

class JournalpostForbiddenException(
    message: String?,
    cause: Throwable? = null,
) : RuntimeException(message, cause)
